# Relatório do aluguel de bicicletas

In [1]:
import numpy as np
import pandas as pd

In [2]:
data = pd.read_csv('bikes.csv')
data.head()

city      dteday  season  yr  mkt  mnth  holiday  weekday  workingday  \
0  city_0  2011-01-01       1   0    0     1        0        6           0   
1  city_0  2011-01-02       1   0    0     1        0        0           0   
2  city_0  2011-01-03       1   0    0     1        0        1           1   
3  city_0  2011-01-04       1   0    0     1        0        2           1   
4  city_0  2011-01-05       1   0    0     1        0        3           1   

   weathersit      temp     atemp       hum  windspeed  stock   cnt  target  
0           2  0.344167  0.363625  0.805833   0.160446    997   929  1257.0  
1           2  0.363478  0.353739  0.696087   0.248539   1231   765  1096.0  
2           1  0.196364  0.189405  0.437273   0.248309   1395  1395  1430.0  
3           1  0.200000  0.212122  0.590435   0.160296   1651  1641   885.0  
4           1  0.226957  0.229270  0.436957   0.186900   2334  1671  1210.0

### Colunas existentes no csv:

In [3]:
data.columns

Index(['city', 'dteday', 'season', 'yr', 'mkt', 'mnth', 'holiday', 'weekday',
       'workingday', 'weathersit', 'temp', 'atemp', 'hum', 'windspeed',
       'stock', 'cnt', 'target'],
      dtype='object')

## Q1: Influência do clima na quantidade de bikes alugadas

#### Média da quantidade de bicicletas alugadas quando o clima está ruim
Clima ruim: Pouca neve ou chuva / Tempestades / Nuvens dispersas e pouca chuva

In [4]:
cnt_bad_weather = data[data['weathersit'] == 3][['cnt','dteday']]
cnt_bad_weather.mean()

cnt    1973.595
dtype: float64

#### Média da quantidade de bicicletas alugadas quando o clima está bom
Clima bom: Tempo limpo / Poucas nuvens / Parcialmente nublado

In [5]:
cnt_good_weather = data[data['weathersit'] == 1][['cnt','dteday']]
cnt_good_weather.mean()

cnt    5009.506793
dtype: float64

In [6]:
pct = (cnt_good_weather.mean() / cnt_bad_weather.mean()) - 1
pct

cnt    1.538265
dtype: float64

### A média de bicicletas alugadas quando o clima está bom é 153% maior do que quando o clima está ruim.

## Q2: Variação da média de bicicletas alugadas durante a semana

#### Média de bicicletas alugadas em cada dia da semana:

In [7]:
data[data['weekday'] == 0][['weekday','cnt']].mean()

weekday       0.000000
cnt        4360.923714
dtype: float64

In [8]:
data[data['weekday'] == 1][['weekday','cnt']].mean()

weekday       1.000
cnt        4478.692
dtype: float64

In [9]:
data[data['weekday'] == 2][['weekday','cnt']].mean()

weekday       2.000000
cnt        4648.866058
dtype: float64

In [10]:
data[data['weekday'] == 3][['weekday','cnt']].mean()

weekday       3.000000
cnt        4687.672163
dtype: float64

In [11]:
data[data['weekday'] == 4][['weekday','cnt']].mean()

weekday       4.00000
cnt        4804.06101
dtype: float64

In [12]:
data[data['weekday'] == 5][['weekday','cnt']].mean()

weekday       5.000000
cnt        4824.593077
dtype: float64

In [13]:
data[data['weekday'] == 6][['weekday','cnt']].mean()

weekday       6.000000
cnt        4686.245048
dtype: float64

#### Cálculo da diferença máxima da média dos dias da semana

In [14]:
valor_min = 4360.923714 
valor_max = 4824.593077
var_max_semana = valor_max - valor_min
var_max_semana

463.6693630000009

### Há uma variação máxima de aproximadamente 464 bikes na média de bicicletas alugadas durante diferente dias da semana.

## Q3: Diferença da quantidade de bikes alugadas nos anos de 2011 e 2012

#### Quantidade total de bikes alugadas nos anos de 2011 e 2012, respectivamente:

In [15]:
total_ano1 = data[data['yr'] == 0]['cnt'].sum()
total_ano1

262768114

In [16]:
total_ano2 = data[data['yr'] == 1]['cnt'].sum()
total_ano2

415750962

#### Cálculo da diferença do número de bicicletas alugadas anualmente (2011 e 2012)

In [17]:
diff = total_ano2 - total_ano1
diff

152982848

### Houve um aumento de aproximadamente 153 milhões de bicicletas alugadas em 2012 (comparado ao ano de 2011)